enhancements/network/ovn-kubernetes-ipsec.md

```diff
@@ -0,0 +1,298 @@
+
+- All Openshift e2e tests should be replicated in a cluster that has been
+  installed with IPsec enabled. This should ensure IPsec has no regressions with
+  respect to a cluster that has not been enabled with IPsec.
+- In order to test any issues with version skew, an IPsec-enabled cluster should
+  be upgraded while active.
+
+Additional CI jobs could potentially be added at a later stage.
+
+- Functional tests
+    - Ensure certificate expiration and rotation happens correctly.
+- Non-functional tests
+    - Performance tests (scale tests)
+    - Endurance tests (traffic generation over a long period)
+    - Recovery tests (what happens when node IPsec components such as pluto or
+      ovs-monitor-ipsec are killed)
+
+It should also be noted that, some upstream OVS/OVN unit and integration tests
+will also need to be developed.
+
+### Graduation Criteria
+
+TBD
+
+#### Examples
+
+##### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+- Performance measurement
+
+##### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+
+##### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+Currently, it is proposed that this feature can only be enabled at cluster
+installation time.
+
+### Version Skew Strategy
+
+It is proposed that this enhancement will not support mixed clusters of RHEL7
+and RHEL8 nodes. This can be supported in a later enhancement.
+
+However, it will need to support version skew within RHEL8 components. The exact
+details of this will be understood, when it is determined how IPsec components
+will be containerized but some considerations are noted below:
+
+- Internode libreswan compatibility
+  - Each nodes' libreswan version will need to be compatible with other nodes'
+    libreswan version
+- Internode OVN/OVS compatibility
+  - Each nodes' OVN/OVS version will need to be compatible with other nodes'
+    OVS/OVN version. While this will generally be covered by OVN Kubernetes as a
+    whole, there may be incompatibilities between IPsec specific
+    subsystems, for example ovsdb schemas entries of newer OVN/OVS versions.
+- ovs-monitor-ipsec/libreswan compatibility
+  - ovs-monitor-ipsec creates libreswan configuration files at runtime.
+    Therefore, the ovs-monitor-ipsec version will need to track libreswan
+    version changes.
+- ovs-monitor-ipsec/ovsdb compatibility
+  - ovs-monitor-ipsec reads ovsdb tables. Therefore, ovs-monitor-ipsec will need
+    to track OVN/OVS versions.
+- libreswan/kernel  compatibility
+  - libreswan will need to be compatible with the underlying Kernel IPsec
+    implementation.
+
+## Implementation History
+
+TBD
+
+## Drawbacks
+
+N/A
+
+## Alternatives
+
+N/A
+
+## Infrastructure Needed [optional]
```

Comment:
```suggestion
## Infrastructure Needed
```

Reply:
Thanks, I will update this on the next revision